### ISSUE TITLE
dialog window title bar are visible in Windows(os)

### ISSUE THREAD

**Issue by ThakeeNathees** [godotengine/godot#37377]
**Godot version:** 307b1b3a5

**OS/device including version:** windows10

**Issue description:**
title bar is visible in dialog windows in Windows(os). also dragging them around makes the window lagging
![windows-title-bar](https://user-images.githubusercontent.com/41085900/77822296-66235700-7117-11ea-8d82-bc319cd1513f.JPG)
![windows-title-bar4](https://user-images.githubusercontent.com/41085900/77822323-a256b780-7117-11ea-8fbf-20c8ad4beb01.png)
![windows-title-bar3](https://user-images.githubusercontent.com/41085900/77822325-a4b91180-7117-11ea-8686-7df93ece37c0.png)
![windows-title-bar2](https://user-images.githubusercontent.com/41085900/77822326-a5ea3e80-7117-11ea-9375-9ae55929cd83.png)

**Comment by Calinou** [comment]
We should probably hide the minimize/maximize buttons for modal dialogs like these. This would free up some space for the window title to display.

**Comment by ThakeeNathees** [comment]
IMO, the Window's title bar and the border should be hidden, and we should draw the title and a close icon.

**Comment by Calinou** [comment]
@ThakeeNathees Indeed, switching to client-side decorations is a long-term plan , but it'll take a fair amount of effort.

**Comment by swarnimarun** [comment]
Why not let some of these be inside the same window? It's not like new Node Dialog needs to be a new Window....

**Comment by RandomShaper** [comment]
I think that small dialogs should also be not resizable. I've always found "unprofessional" that a dialog with very little content can be resized so it looks weird, as if the UI was unpolished.

**Comment by reduz** [comment]
I don't think this should be classified as a bug, and its a bit difficult to infer which dialogs should be resizable and which should not. This is not something that could be done automatically in the AcceptDialog base class, but something that someone has to go and disable in every dialog that they don't think should be resizable.

**Comment by KoBeWi** [comment]
I think the title bars are ok (and the dragging lag is gone for some time now), but we should at least make sure that whole title is displayed.
![image](https://user-images.githubusercontent.com/2223172/178156299-347a41f8-b107-4aba-aa83-8d4c2ebbd9f5.png)

> We should probably hide the minimize/maximize buttons for modal dialogs like these.

This can be achieved with the `unresizable` flag, but this makes the window unresizable (bruh).

**Comment by Calinou** [comment]
I agree AcceptDialogs should generally not be resizable, but we should make sure we fix the issue where editor dialogs occasionally spawn with a size that's too large before making them unresizable.

**Comment by KoBeWi** [comment]
We can hide the buttons, but still keep the window resizable. Not supported right now, but it's possible.
